docs/content/preview/architecture/docdb-replication/raft.md

```diff
@@ -0,0 +1,74 @@
+
+## Log replication
+
+The leader is responsible for replicating log entries to the followers. When a client sends a request to the leader, the leader appends the request to its local log and then sends the log entry to the followers. The followers then append the log entry to their local logs and send an acknowledgment back to the leader. Once the leader receives a majority of acknowledgments, it considers the log entry to be committed and can apply it to its state machine.
+
+## Replication of the write operation
+
+All write operations are handled by the tablet leader. These are the squence of operations that happen during the replication of a write.
+
+1. The leader first takes a lock on the key using a local in-memory lock manager and reads data if necessary (for read-modify-write or conditional update operations). Note that this locking mechanism does not exist on followers.
+1. The batch of changes to be written to [DocDB](../../docdb/) is prepared. This write batch is very close to the final set of RocksDB key-value pairs to be written, only lacking the final [hybrid timestamp](../../transactions/transactions-overview#hybrid-logical-clocks) at the end of each key.
+1. The leader appends the batch to its Raft log and chooses a [hybrid timestamp](../../transactions/transactions-overview#hybrid-logical-clocks) for the write operation.
+1. The data is replicated via Raft to its peers.
+1. After receiving a `replication successful` callback from the Raft subsystem, the leader applies the write batch to its local DocDB.
+1. Responds with success to the user.
+1. The next update from the leader notifies followers that the entry has been committed, and the followers apply the write batch to their RocksDB instances.
+
+## Leader election
+
+When the system starts up, each node is assigned an initial election term, which is typically a non-negative integer. When a  follower does not receive a heartbeat within the election timeout period it becomes a candidate to start a new election term and attempt to become the new Leader. Each node votes for a candidate in a particular election term. The candidate that receives votes from a majority of the nodes is elected as the new leader. Whenever a new leader is elected, the election term is incremented. This ensures that each term has a unique identifier, and it helps the nodes in the system differentiate between the various terms.
+
+If a node receives a request from a leader with a stale term (i.e., a term that is lower than the node's current term), the node will reject the request and inform the leader of the newer term. This helps prevent outdated leaders from making changes to the replicated log. RAFT ensures that the replicated log remains consistent across the nodes by requiring that all log entries be associated with a specific term. This means that a node will only accept log entries that are from the current term or a previous term
```

Comment:
ensures is too strong here as a lot of other things are required as well.
the last sentence is just wrong.  A node that has been down for a while when coming up will get caught up with log records for everything it missed, which can include log records from quite a few different terms.